python/ModelTools.py

```diff
@@ -278,12 +259,37 @@ def doNuisances(self):
             self.doObj("nuisancePdf", "PROD", ",".join(["%s_Pdf" % n for (n,nf,p,a,e) in self.DC.systs]))
             self.doSet("globalObservables", ",".join(globalobs))
 	
+    def doNuisancesGroups(self):
+        # Prepare a dictionary of which group a certain nuisance belongs to
+        groupsFor = {}
+        existingNuisanceNames = tuple(set([syst[0] for syst in self.DC.systs]+self.DC.flatParamNuisances.keys()))
```

Comment:
@nucleosynthesis proposed to also add `self.DC.discretes` to this expanded list.
`discretes` is filled in https://github.com/cms-analysis/HiggsAnalysis-CombinedLimit/blob/slc6-root5.34.17/python/DatacardParser.py#L180-L183